pkg/plugin/v3/scaffolds/internal/templates/config/kdefault/auth_proxy_patch.go

```diff
@@ -36,14 +40,25 @@ func (f *AuthProxyPatch) SetTemplateDefaults() error {
 		f.Path = filepath.Join("config", "default", "manager_auth_proxy_patch.yaml")
 	}
 
-	f.TemplateBody = kustomizeAuthProxyPatchTemplate
+	args := argsFragment
+	if f.ComponentConfig {
+		args = ""
+	}
+
+	f.TemplateBody = fmt.Sprintf(kustomizeAuthProxyPatchTemplate, args)
```

Comment:
Same as in `main`, template inside the template body with `{{ if .ComponentConfig }}` instead of in `SetTemplateDefaults`.